<?xml version="1.0" encoding="utf-8"?>
<robot name="pathfinder">
    <!-- Inertia box Ixx = 1/12 * (ly*2 + lh*2)-->
    <link name="odom"/>

    <joint name="cart_x" type="prismatic">
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="odom"/>
        <child link="cart_link"/>
        <axis xyz="1 0 0"/>
        <limit effort="1000" lower="-30.14" upper="30.14" velocity="2.0"/>
    </joint>

    <link name="cart_link">
        <inertial>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <mass value="1.0"/>
            <inertia ixx="0.001667" ixy="0.0" ixz="0.0" iyy="0.0841667" iyz="0.0" izz="0.084167"/>
        </inertial>
        <visual>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <geometry>
                <box size="0.03 0.03 0.03"/>
            </geometry>
        </visual>
    </link>

    <joint name="cart_y" type="prismatic">
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <parent link="cart_link"/>
            <child link="cart_link2"/>
            <axis xyz="0 1 0"/>
            <limit effort="1000" lower="-30.14" upper="30.14" velocity="2.0"/>
    </joint>

    <link name="cart_link2">
            <inertial>
                <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
                <mass value="1.0"/>
                <inertia ixx="0.001667" ixy="0.0" ixz="0.0" iyy="0.0841667" iyz="0.0" izz="0.084167"/>
            </inertial>
            <visual>
                <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
                <geometry>
                    <box size="0.03 0.03 0.03"/>
                </geometry>
            </visual>
    </link>
    <joint name="cart_theta" type="continuous">
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <parent link="cart_link2"/>
            <child link="cart_link3"/>
            <axis xyz="0 0 1"/>
            <limit effort="1000" velocity="1.0"/>
    </joint>
        <link name="cart_link3">
                <inertial>
                    <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
                    <mass value="1.0"/>
                    <inertia ixx="0.001667" ixy="0.0" ixz="0.0" iyy="0.0841667" iyz="0.0" izz="0.084167"/>
                </inertial>
                <visual>
                    <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
                    <geometry>
                        <box size="0.3 0.3 0.3"/>
                    </geometry>
                </visual>
        </link>

        <joint name="fixed" type="fixed">
            <parent link="odom"/>
            <child link="obstacle"/>
            <origin xyz="14.0 2.0 0.0" rpy="0 0 0"/>
        </joint>

        <link name="obstacle">
                <inertial>
                    <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
                    <mass value="1.0"/>
                    <inertia ixx="0.001667" ixy="0.0" ixz="0.0" iyy="0.0841667" iyz="0.0" izz="0.084167"/>
                </inertial>
                <visual>
                    <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
                    <geometry>
                        <cylinder length="2" radius="2"/>
                    </geometry>
                </visual>
        </link>

</robot>